packages/expo-print/android/src/main/java/expo/modules/print/FileUtils.kt

```diff
@@ -27,4 +36,37 @@ object FileUtils {
     val filename = UUID.randomUUID().toString()
     return directory.toString() + File.separator + filename + extension
   }
+
+  @Throws(IOException::class)
+  fun copyToOutputStream(destination: ParcelFileDescriptor, callback: PrintDocumentAdapter.WriteResultCallback, input: InputStream) {
+    FileOutputStream(destination.fileDescriptor).use {
+      val buf = ByteArray(1024)
+      var bytesRead = input.read(buf)
+      while (bytesRead > 0) {
+        it.write(buf, 0, bytesRead)
+        bytesRead = input.read(buf)
+      }
+      callback.onWriteFinished(arrayOf(PageRange.ALL_PAGES))
+    }
+  }
```

Comment:
I think you can simplify this function using https://stackoverflow.com/a/35529070/13361748.